pybamm/install_odes.py

```diff
@@ -16,6 +17,9 @@
 except ModuleNotFoundError:
     NO_WGET = True
 
+# Build in parallel wherever possible
+os.environ["CMAKE_BUILD_PARALLEL_LEVEL"] = str(cpu_count())
```

Comment:
One small doubt, can we safely say that this PR closes #3646 or there may be more scope possible of this implementation.

Reply:
I have done that, but with this I realise that I haven't added this environment variable for the Windows build. Will do that in a moment

Reply:
And done. Doesn't seem to have helped much though, but it's the first time I saw five Windows wheels being compiled in ~24 minutes, so we should be better with it rather than without. 

https://github.com/agriyakhetarpal/PyBaMM/actions/runs/7431035977

Reply:
Will leave it up to @Saransh-cpp to merge as per his convenience since he has to cherry-pick these changes as well